### PR TITLE
Some unit tests for the tools

### PR DESCRIPTION
**Pull request by gzm0**
Partial #1606

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

File: tools/shared/src/test/scala/org/scalajs/core/tools/jsdep/ManifestFiltersTest.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  private def mkManifest(module: String, deps: String*) = {
+    new JSDependencyManifest(new Origin(module, "compile"),
+        deps.map(new JSDependency(_)).toList, requiresDOM = false,
+        compliantSemantics = Nil)
+  }
+
+  @Test
+  def reinterpretResourceNamesSimple(): Unit = {
+    val filter = reinterpretResourceNames("lib-b")(
+      "jquery.js" -> "2.1.3/jquery.js")
```
Comment (sjrd) [inline tools/shared/src/test/scala/org/scalajs/core/tools/jsdep/ManifestFiltersTest.scala]: 4 spaces

**Comment by sjrd** [comment]
That's all.

**Comment by sjrd** [comment]
Ping @gzm0 ;-)

File: tools/shared/src/main/scala/org/scalajs/core/tools/jsdep/ComplianceRequirement.scala [hunk 2]
```diff
@@ -4,9 +4,30 @@ import org.scalajs.core.tools.sem.Semantics
 
 /** Expresses a requirement for a given semantic to be compliant */
 final class ComplianceRequirement(
-    val semantics: String, val origins: List[Origin])
+    val semantics: String, val origins: List[Origin]) {
+  override def toString(): String =
+      s"ComplianceRequirement($semantics, origins = [${origins.mkString(", ")}])"
```
Comment (sjrd) [inline tools/shared/src/main/scala/org/scalajs/core/tools/jsdep/ComplianceRequirement.scala]: And now I see this one ^^ 2 spaces here.

**Comment by gzm0** [comment]
Updated.

**Comment by sjrd** [comment]
LGTM :)

**Pull request merged**